test/LogTest.php

```diff
@@ -0,0 +1,208 @@
+
+        $level = $opusLog->getLevel();
+
+        $this->assertContains($debugMessage, $content);
+        $this->assertSame(\Zend_Log::DEBUG, $level);
+    }
+
+    public function testSetLevelForFilterDisabling()
+    {
+        $opusLog = $this->getOpusLog();
+        $opusLog->setLevel(null);
+
+        $debugMessage = 'Debug level message from testSetPriorityWithNullPriority';
+        $opusLog->debug($debugMessage);
+
+        $content = $this->readLog();
+
+        $this->assertContains($debugMessage, $content);
+    }
+
+    public function testSetLevelToNullHavingCustomLevels()
+    {
+        $opusLog = $this->getOpusLog();
+        $opusLog->addPriority('TEST', 8);
+        $opusLog->setLevel(null);
+        $opusLog->addPriority('TLEVEL', 9);
+
+        $testMessage = 'Test level message';
+        $opusLog->test($testMessage);
+
+        $tlevelMessage = 'Tlevel message';
+        $opusLog->tlevel($tlevelMessage);
+
+        $content = $this->readLog();
+
+        $this->assertContains($testMessage, $content);
+        $this->assertContains($tlevelMessage, $content);
+    }
+
+    public function testSetLevelNotEffectingOtherFilters()
+    {
+        $opusLog = $this->getOpusLog();
+
+        $filter = new \Zend_Log_Filter_Priority(\Zend_Log::WARN);
+        $opusLog->addFilter($filter);
+
+        $infoMessage = 'Info level Message';
+        $opusLog->info($infoMessage);
+
+        $opusLog->setLevel(\Zend_Log::NOTICE);
+        $noticeMessage = 'Notice Level Message';
+        $opusLog->notice($noticeMessage);
+
+        $opusLog->setLevel(\Zend_Log::ERR);
+        $errorMessage = 'Error level Message';
+        $opusLog->err($errorMessage);
+
+        $content = $this->readLog();
+
+        $this->assertNotContains($infoMessage, $content);
+        $this->assertNotContains($noticeMessage, $content);
+        $this->assertContains($errorMessage, $content);
+    }
+
+    public function testSetLevelNegativePriority()
+    {
+        $opusLog = $this->getOpusLog();
+
+        $this->setExpectedException(\InvalidArgumentException::class, 'Priority should be of Integer type and cannot be negative');
+
+        $opusLog->setLevel(-1);
+    }
+
+    public function testSetLevelNotIntPriority()
+    {
+        $opusLog = $this->getOpusLog();
+
+        $this->setExpectedException(\InvalidArgumentException::class, 'Priority should be of Integer type and cannot be negative');
+
+        $opusLog->setLevel('TestPriority');
+    }
+
+    public function testGetLevel()
+    {
+        $opusLog = $this->getOpusLog();
+        $priority = $opusLog->getLevel();
+
+        $this->assertEquals(\Zend_Log::INFO, $priority);
+    }
+
+    public function testGetLevelAfterSetPriority()
+    {
+        $opusLog = $this->getOpusLog();
+        $opusLog->setLevel(\Zend_Log::DEBUG);
+        $priority = $opusLog->getLevel();
+
+        $this->assertEquals(\Zend_Log::DEBUG, $priority);
+    }
+
+    public function testGetLevelOnNullPriority()
```

Comment:
We want to have tests covering sunshine and rain conditions for all aspects of the class, but I have doubts about this test. Maybe it is a naming problem, but it seems to be part of another test.

Reply:
Changed the name to `testGetLevelReturnsNull()`.